python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -0,0 +1,308 @@
+                            ),
+                        )
+            if times_column is not None:
+                col = (
+                    col.astype(dtype="int64")
+                    .binary_operator(binop="add", rhs=times_column)
+                    .astype(
+                        dtype="datetime64[s]"
+                        if np.datetime_data(col.dtype)[0] == "s"
+                        else "datetime64[ns]"
+                    )
+                )
+            return cudf.Series(col, index=arg.index)
+        elif isinstance(arg, cudf.Index):
+            col = arg._values
+            col = _process_col(
+                col=col,
+                unit=unit,
+                dayfirst=dayfirst,
+                infer_datetime_format=infer_datetime_format,
+                format=format,
+            )
+            return as_index(col, name=arg.name)
+        elif isinstance(arg, cudf.Series):
+            col = arg._column
+            col = _process_col(
+                col=col,
+                unit=unit,
+                dayfirst=dayfirst,
+                infer_datetime_format=infer_datetime_format,
+                format=format,
+            )
+            return cudf.Series(col, index=arg.index, name=arg.name)
+        else:
+            col = column.as_column(arg)
+            col = _process_col(
+                col=col,
+                unit=unit,
+                dayfirst=dayfirst,
+                infer_datetime_format=infer_datetime_format,
+                format=format,
+            )
+
+            if is_scalar(arg):
+                return col[0]
+            else:
+                return as_index(col)
+    except Exception as e:
+        if errors == "raise":
+            raise e
+        elif errors == "warn":
+            import traceback
+
+            tb = traceback.format_exc()
+            warnings.warn(tb)
+        elif errors == "ignore":
+            pass
+        elif errors == "coerce":
+            return np.datetime64("nat", "ns" if unit is None else unit)
+        return arg
+
+
+def _process_col(col, unit, dayfirst, infer_datetime_format, format):
+    if col.dtype.kind == "M":
+        return col
+
+    if col.dtype.kind in ("f"):
+        if unit not in (None, "ns"):
+            factor = as_scalar(
+                column.datetime._numpy_to_pandas_conversion[unit]
+            )
+            col = col.binary_operator(binop="mul", rhs=factor)
+        col = col.as_datetime_column(dtype="datetime64[ns]")
+    if col.dtype.kind in ("i"):
+        if unit in ("D"):
+            factor = as_scalar(
+                column.datetime._numpy_to_pandas_conversion[unit]
+                / column.datetime._numpy_to_pandas_conversion["s"]
+            )
+            col = col.binary_operator(binop="mul", rhs=factor)
+            col = col.as_datetime_column(dtype="datetime64[s]")
+        else:
+            col = col.as_datetime_column(dtype="datetime64[" + unit + "]")
```

Comment:
Would use a lookup table or different approach as opposed to doing the string concatenation like this.